app/src/main/java/com/nervousfish/nervousfish/LoginActivity.java

```diff
@@ -24,46 +22,33 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setContentView(R.layout.login);
 
-        mEmail = (EditText) findViewById(R.id.email);
         mPassword = (EditText) findViewById(R.id.password);
 
         View submitButton = findViewById(R.id.submit);
         submitButton.setOnClickListener(new View.OnClickListener() {
 
             @Override
             public void onClick(View v) {
-                validateFields();
-
-                if (mEmail.getError() == null && mPassword.getError() == null) {
+                if (mPassword.getError() == null) {
                     validateAccount();
                 }
             }
         });
     }
 
-    private void validateFields() {
-        if (!android.util.Patterns.EMAIL_ADDRESS.matcher(mEmail.getText().toString()).matches()) {
-            mEmail.setError(getString(R.string.msg_email_error));
-        } else {
-            mEmail.setError(null);
-        }
-
-        if (mPassword.getText().toString().isEmpty()) {
-            mPassword.setError(getString(R.string.msg_password_error));
-        } else {
-            mPassword.setError(null);
-        }
-    }
-
     private void validateAccount() {
         if (mError == null) {
             mError = findViewById(R.id.error);
         }
 
-        if (!mEmail.getText().toString().equals(DEMO_EMAIL) || !mPassword.getText().toString().equals(DEMO_PASSWORD)) {
+        // TODO: have a real password check
+        boolean wrongPassword = !mPassword.getText().toString().isEmpty();
```

Comment:
`wrongPassword` seems like a wierd name for this variable to me, given the value you're assigning to it... Right now the password is wrong if the string is not empty?

How about:
```java
boolean skipPassword = mPassword.getText().toString().isEmpty();
if (skipPassword == true) {
  // Next activity
} else {
  // 1. Check password
  // 2. Next activity (if the password is correct)
}
```